lib/js_locale_helper.rb

```diff
@@ -1,30 +1,94 @@
 module JsLocaleHelper
 
-  def self.output_locale(locale, translations = nil)
-    current_locale = I18n.locale
-    I18n.locale = locale.to_sym
+  def self.load_translations(locale)
+    @loaded_translations ||= HashWithIndifferentAccess.new
+    @loaded_translations[locale] ||= begin
+      locale_str = locale.to_s
+
+      # load default translations
+      translations = YAML::load(File.open("#{Rails.root}/config/locales/client.#{locale_str}.yml"))
+      # load plugins translations
+      plugin_translations = {}
+      Dir["#{Rails.root}/plugins/*/config/locales/client.#{locale_str}.yml"].each do |file|
+        plugin_translations.deep_merge! YAML::load(File.open(file))
+      end
+
+      # merge translations (plugin translations overwrite default translations)
+      translations[locale_str]['js'].deep_merge!(plugin_translations[locale_str]['js']) if translations[locale_str] && plugin_translations[locale_str] && plugin_translations[locale_str]['js']
+
+      # We used to split the admin versus the client side, but it's much simpler to just
+      # include both for now due to the small size of the admin section.
```

Comment:
Wait, does this mean even in production users will get the admin section even as regular users? I'm not sure if I support that. What is the size difference?